# First Analysis: Cuisine_rating.csv

Standard first-pass inspection: structure, data quality, distributions, business context, and next questions.

In [1]:
import pandas as pd
pd.set_option('display.max_columns', None)
pd.set_option('display.width', 200)

df = pd.read_csv('Cuisine_rating.csv')
df.columns = [c.strip() for c in df.columns]
print('Shape:', df.shape)

Shape: (200, 15)


## 1-2. Dataset overview

In [2]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 15 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   User ID         200 non-null    int64  
 1   Area code       200 non-null    int64  
 2   Location        200 non-null    str    
 3   Gender          200 non-null    str    
 4   YOB             200 non-null    int64  
 5   Marital Status  200 non-null    str    
 6   Activity        200 non-null    str    
 7   Budget          200 non-null    int64  
 8   Cuisines        200 non-null    str    
 9   Alcohol         200 non-null    str    
 10  Smoker          200 non-null    str    
 11  Food Rating     200 non-null    int64  
 12  Service Rating  200 non-null    int64  
 13  Overall Rating  200 non-null    float64
 14  Often A S       200 non-null    str    
dtypes: float64(1), int64(6), str(8)
memory usage: 23.6 KB


In [3]:
df.head(10)

,User ID,Area code,Location,Gender,YOB,Marital Status,Activity,Budget,Cuisines,Alcohol,Smoker,Food Rating,Service Rating,Overall Rating,Often A S
0,1,153,"Upper East Side,NY",Female,2006,Single,Professional,3,Japanese,Never,Never,5,4,4.5,No
1,2,123,"St. George,NY",Female,1991,Married,Student,3,Indian,Never,Socially,1,1,1.0,No
2,3,122,"Upper West Side,NY",Male,1977,Single,Student,5,Seafood,Often,Often,5,5,5.0,Yes
3,4,153,"Upper East Side,NY",Female,1956,Married,Professional,5,Japanese,Never,Socially,3,1,2.0,No
4,5,129,"Central Park,NY",Male,1997,Single,Student,4,Filipino,Socially,Never,2,4,3.0,No
5,6,111,"China Town, NY",Male,1995,Single,Student,5,Seafood,Never,Never,5,1,3.0,No
6,7,111,"China Town, NY",Female,1977,Married,Professional,5,Indian,Often,Socially,1,4,2.5,No
7,8,153,"Upper East Side,NY",Female,2003,Single,Professional,3,Filipino,Often,Often,5,2,3.5,Yes
8,9,107,"Riverdale,NY",Male,1965,Single,Student,5,Italian,Socially,Socially,3,3,3.0,No
9,10,129,"Central Park,NY",Male,1995,Single,Student,4,Chinese,Often,Often,5,2,3.5,Yes


## 3. Missing values

In [4]:
missing = pd.DataFrame({
    'missing_count': df.isna().sum(),
    'missing_pct': (df.isna().mean() * 100).round(2)
}).sort_values('missing_count', ascending=False)
missing

,missing_count,missing_pct
User ID,0,0.0
Area code,0,0.0
Location,0,0.0
Gender,0,0.0
YOB,0,0.0
Marital Status,0,0.0
Activity,0,0.0
Budget,0,0.0
Cuisines,0,0.0
Alcohol,0,0.0


No missing values in any column.

## 4. Descriptive statistics

In [5]:
df.describe(include='all')

,User ID,Area code,Location,Gender,YOB,Marital Status,Activity,Budget,Cuisines,Alcohol,Smoker,Food Rating,Service Rating,Overall Rating,Often A S
count,200.000000,200.000000,200,200,200.000000,200,200,200.000000,200,200,200,200.000000,200.000000,200.000000,200
unique,NaN,NaN,10,2,NaN,3,2,NaN,7,3,3,NaN,NaN,NaN,2
top,NaN,NaN,"St. George,NY",Male,NaN,Single,Student,NaN,Japanese,Never,Socially,NaN,NaN,NaN,No
freq,NaN,NaN,46,118,NaN,100,120,NaN,36,88,71,NaN,NaN,NaN,174
mean,100.500000,141.060000,NaN,NaN,1984.830000,NaN,NaN,3.815000,NaN,NaN,NaN,3.220000,3.230000,3.225000,NaN
std,57.879185,26.130257,NaN,NaN,16.809339,NaN,NaN,1.056578,NaN,NaN,NaN,1.411226,1.526022,1.079445,NaN
min,1.000000,101.000000,NaN,NaN,1955.000000,NaN,NaN,1.000000,NaN,NaN,NaN,1.000000,1.000000,1.000000,NaN
25%,50.750000,123.000000,NaN,NaN,1971.000000,NaN,NaN,3.000000,NaN,NaN,NaN,2.000000,2.000000,2.500000,NaN
50%,100.500000,135.000000,NaN,NaN,1987.000000,NaN,NaN,4.000000,NaN,NaN,NaN,3.000000,3.000000,3.000000,NaN
75%,150.250000,158.000000,NaN,NaN,2000.000000,NaN,NaN,5.000000,NaN,NaN,NaN,5.000000,5.000000,4.000000,NaN


## 5. Cardinality

In [6]:
df.nunique(dropna=False).sort_values()

Gender              2
Activity            2
Often A S           2
Marital Status      3
Alcohol             3
Smoker              3
Budget              5
Food Rating         5
Service Rating      5
Cuisines            7
Overall Rating      9
Location           10
YOB                46
Area code          65
User ID           200
dtype: int64

## 6. Data quality checks

In [7]:
print('Duplicate rows:', df.duplicated().sum())
print('Duplicate User IDs:', df['User ID'].duplicated().sum())

Duplicate rows: 0
Duplicate User IDs: 0


In [8]:
for col in df.select_dtypes(include='object').columns:
    print(f'--- {col} ---')
    print(df[col].value_counts(dropna=False))
    print()

--- Location ---
Location
St. George,NY         46
Upper East Side,NY    30
Riverdale,NY          28
Central Park,NY       24
China Town, NY        22
Market City, NY       20
Upper West Side,NY    18
Central Park,ny        8
Market City, MY        2
Cedar Hill, NY         2
Name: count, dtype: int64

--- Gender ---
Gender
Male      118
Female     82
Name: count, dtype: int64

--- Marital Status ---
Marital Status
Single      100
Married      86
Divorced     14
Name: count, dtype: int64

--- Activity ---
Activity
Student         120
Professional     80
Name: count, dtype: int64

--- Cuisines ---
Cuisines
Japanese    36
Filipino    34
French      34
Indian      32
Chinese     24
Seafood     22
Italian     18
Name: count, dtype: int64

--- Alcohol ---
Alcohol
Never       88
Often       61
Socially    51
Name: count, dtype: int64

--- Smoker ---
Smoker
Socially    71
Often       70
Never       59
Name: count, dtype: int64

--- Often A S ---
Often A S
No     174
Yes     26
Name: count, dty

/var/folders/sz/3_np9r_91db6_dhc9819tkwh0000gn/T/ipykernel_80573/2576641868.py:1: Pandas4Warning: For backward compatibility, 'str' dtypes are included by select_dtypes when 'object' dtype is specified. This behavior is deprecated and will be removed in a future version. Explicitly pass 'str' to `include` to select them, or to `exclude` to remove them and silence this warning.
See https://pandas.pydata.org/docs/user_guide/migration-3-strings.html#string-migration-select-dtypes for details on how to write code that works with pandas 2 and 3.
  for col in df.select_dtypes(include='object').columns:


**Flagged issue: `Location` is fragmented by inconsistent formatting.**

- `Central Park,NY` (24) vs `Central Park,ny` (8) — casing inconsistency, same location.
- `Market City, NY` (20) vs `Market City, MY` (2) — looks like a data-entry typo (NY → MY), not a real second city.
- `China Town, NY` has a leading space after the comma that other entries don't.

10 distinct values likely represent ~8 real locations. Any location-level ranking below is provisional until this is cleaned.

## 7. Numerical distributions and relationships

In [9]:
# Check whether Overall Rating is an independent measurement or derived
calc_overall = (df['Food Rating'] + df['Service Rating']) / 2
print('Overall Rating == mean(Food, Service) for all rows?', (calc_overall - df['Overall Rating']).abs().lt(1e-9).all())

Overall Rating == mean(Food, Service) for all rows? True


`Overall Rating` is exactly `(Food Rating + Service Rating) / 2` for every row — it is derived, not an independent signal. Treat Food and Service as the two real outcome variables.

In [10]:
df['Age'] = 2026 - df['YOB']
df[['Budget', 'Food Rating', 'Service Rating', 'Overall Rating', 'Age']].corr(numeric_only=True).round(2)

,Budget,Food Rating,Service Rating,Overall Rating,Age
Budget,1.00,0.06,-0.14,-0.06,0.07
Food Rating,0.06,1.00,0.08,0.71,-0.04
Service Rating,-0.14,0.08,1.00,0.76,-0.04
Overall Rating,-0.06,0.71,0.76,1.00,-0.06
Age,0.07,-0.04,-0.04,-0.06,1.00


Budget and Age show essentially no correlation with any rating (|r| < 0.15). Food Rating and Service Rating are only weakly correlated with each other (0.08) — a diner's food experience and service experience move largely independently.

In [11]:
df['Overall Rating'].value_counts().sort_index()

Overall Rating
1.0     6
1.5    11
2.0    21
2.5    32
3.0    34
3.5    35
4.0    26
4.5     6
5.0    29
Name: count, dtype: int64

## 8. Business context: Cuisines as "product", Location as "region"

This is a restaurant-dining survey, not a revenue/profit dataset. There is no company, revenue, or cost data. `Overall Rating` is treated as the business KPI in place of profit; `Cuisines` as the product dimension; `Location` as the region dimension.

## 9. Cuisine performance

In [12]:
cuisine_perf = df.groupby('Cuisines').agg(
    n=('User ID', 'count'),
    avg_overall=('Overall Rating', 'mean'),
    avg_food=('Food Rating', 'mean'),
    avg_service=('Service Rating', 'mean'),
).sort_values('avg_overall', ascending=False).round(2)
cuisine_perf

,n,avg_overall,avg_food,avg_service
Cuisines,,,,
Japanese,36,3.60,3.86,3.33
Chinese,24,3.27,3.46,3.08
Italian,18,3.19,3.22,3.17
Indian,32,3.17,2.78,3.56
Seafood,22,3.14,3.23,3.05
French,34,3.10,3.29,2.91
Filipino,34,3.04,2.71,3.38


Japanese rates highest overall (3.60); Filipino lowest (3.04). Notable: **Indian and Filipino have the two lowest Food scores but above-average Service scores** — food and service diverge in opposite directions for these cuisines, which the single blended `Overall Rating` masks.

## 10. Location performance (provisional — raw, uncleaned)

In [13]:
location_perf = df.groupby('Location').agg(
    n=('User ID', 'count'),
    avg_overall=('Overall Rating', 'mean'),
).sort_values('avg_overall', ascending=False).round(2)
location_perf

,n,avg_overall
Location,,
"Market City, NY",20,3.72
"Central Park,ny",8,3.69
"Cedar Hill, NY",2,3.50
"Central Park,NY",24,3.44
"China Town, NY",22,3.16
"Upper West Side,NY",18,3.14
"St. George,NY",46,3.12
"Riverdale,NY",28,3.05
"Upper East Side,NY",30,3.02


`St. George,NY` is the largest location by volume (46 records) but rates below average (3.12). This table will shift once the casing/typo issues from the data-quality section are fixed.

## 11. Segment analysis

In [14]:
for col in ['Marital Status', 'Activity', 'Gender', 'Alcohol', 'Smoker', 'Often A S']:
    print(f'--- {col} ---')
    print(df.groupby(col)['Overall Rating'].agg(['count', 'mean']).round(2))
    print()

--- Marital Status ---
                count  mean
Marital Status             
Divorced           14  4.54
Married            86  2.87
Single            100  3.34

--- Activity ---
              count  mean
Activity                 
Professional     80  3.44
Student         120  3.08

--- Gender ---
        count  mean
Gender             
Female     82  3.34
Male      118  3.15

--- Alcohol ---
          count  mean
Alcohol              
Never        88  2.97
Often        61  3.39
Socially     51  3.47

--- Smoker ---
          count  mean
Smoker               
Never        59  2.74
Often        70  3.89
Socially     71  2.98

--- Often A S ---
           count  mean
Often A S             
No           174  3.09
Yes           26  4.10



- **Marital Status**: Divorced diners rate much higher (4.54, n=14) than Married (2.87, n=86) or Single (3.34, n=100). Direction is clear but the Divorced group is small — worth confirming it holds up, not a settled finding.
- **Smoker**: "Often" smokers average 3.89 vs "Never" 2.74 and "Socially" 2.98 — a large, non-monotonic gap with no obvious causal story. Flagged as curious, not explained.
- **Alcohol**: cleaner gradient — Never (2.97) < Often (3.39) < Socially (3.47).
- **`Often A S`**: column name is ambiguous (possibly "Often Asks for Substitutions/Special"); Yes rates much higher (4.10) than No (3.09). Confirm the column's actual meaning before relying on it.

In [15]:
df.groupby('Budget')['Overall Rating'].agg(['count', 'mean']).round(2)

,count,mean
Budget,,
1,10,3.85
2,4,3.50
3,61,3.14
4,63,3.15
5,62,3.27


Budget shows no clear pattern (flat 3.14–3.27 across tiers 3–5); the Budget=1 bucket looks higher (3.85) but n=10, likely noise.

## Assumptions

- Age computed as `2026 − YOB` (no explicit "today" reference in the data).
- Treated `Cuisines` as the product dimension and `Location` as the region dimension, since there is no revenue/profit/company data.
- Location casing/typo issues were **not** normalized before computing the location table — those numbers are provisional.

## Questions this dataset can answer next

1. Does the Food-vs-Service divergence for Indian/Filipino cuisine hold after normalizing location, or is it confounded by which locations serve those cuisines?
2. Is the Divorced-diner rating gap robust, or an artifact of small n (14)?
3. What does `Often A S` actually mean, and does it meaningfully predict rating once other segments are controlled for?
4. Do Budget tier and Activity (Student vs Professional) interact?
5. After cleaning `Location`, which locations are genuinely top/bottom performers, and by how much?
6. Since Overall = mean(Food, Service), is it more useful to analyze Food and Service as two separate outcomes rather than collapsing them?

## Final summary

**FIRST ANALYSIS SUMMARY**

Dataset: 200 rows × 15 columns, no missing values, no duplicates.

Data quality: ⚠️ `Location` fragmented by casing/typo (10 values → ~8 real locations). ⚠️ `Overall Rating` is derived (mean of Food + Service), not independent. ⚠️ `Often A S` has an ambiguous, unconfirmed meaning.

Business (rating as KPI): mean Overall Rating 3.23 (std ~1.08), spread fairly evenly from 1 to 5.

Cuisines: Japanese rates highest (3.60); Filipino lowest (3.04). Indian and Filipino show an unusual food-low / service-high split.

Locations: `Market City, NY` and `Central Park` lead (provisional); `St. George,NY`, the largest location by volume, rates below average.

Most interesting findings:
- Food and Service ratings are largely independent of each other (r=0.08).
- Divorced diners and "Often" smokers rate notably higher than other segments — real patterns in the data, not explained causally, and based on modest sample sizes.
- Budget and Age have no meaningful relationship with satisfaction.

Recommended next analysis:
1. Clean `Location` and re-run the location ranking.
2. Investigate the Food/Service divergence by cuisine, controlling for location.
3. Confirm the meaning of `Often A S` and re-test its association with rating.